src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -537,6 +544,86 @@ getModule df n = do
   return (pkg, T.pack $ moduleNameString $ moduleName m)
 
 -- ---------------------------------------------------------------------
+-- TODO: there has to be a simpler way, using the appropriate GHC internals
+-- |Find the Id for a given Name.
+-- Requires an already TypecheckedModule.
+-- A TypecheckedModule can be obtained by using the functions 
+-- @ifCachedModuleAndData@ or @withCachedModuleAndData@.
+--
+-- Function is copied from @HaRe/src/Language/Haskell/Refact/Utils/TypeUtils.hs:2954@. 
+findIdForName :: TypecheckedModule -> Name -> IdeM (Maybe Id)
+findIdForName tm n = do
+  let t = GHC.tm_typechecked_source tm
+  let r = SYB.something (SYB.mkQ Nothing worker) t
+      worker (i :: GHC.Id) | nameUnique n == varUnique i = Just i
+      worker _ = Nothing
+  return r
+
+-- ---------------------------------------------------------------------
+
+-- | Get the type for a name.
+-- Requires an already TypecheckedModule.
+-- A TypecheckedModule can be obtained by using the functions 
+-- @ifCachedModuleAndData@ or @withCachedModuleAndData@.
+-- 
+-- Returns the type of a variable or a sum type constructor.
+-- 
+-- Function is taken from @HaRe/src/Language/Haskell/Refact/Utils/TypeUtils.hs:2966@.
+getTypeForName' :: TypecheckedModule -> Name -> IdeM (Maybe Type)
```

Comment:
It might be worth testing, whether `getTypeForName'` could replace `getTypeForName`. 
I had to write this function, since it was able to find the type of a given variable, which did not work with `getTypeForName`.